IdentityCore/src/parameters/MSIDRequestParameters.h

```diff
@@ -77,6 +77,8 @@
 
 - (BOOL)validateParametersWithError:(NSError **)error;
 
+- (void)setRequestV1IdToken:(BOOL)requestV1IdToken;
```

Comment:
would it help if we used different token request provider, which looks up different ID token type?
e.g. subclass MSIDDefaultSilentTokenRequest, which looks up different ID token for ADAL requests.

Reply:
Currently we are using `MSIDRequestParameters`/`MSIDConfiguration` object to pass the request-related information all the way down to the cache accessor.

Seems subclassing `MSIDDefaultTokenRequestProvider`, `MSIDDefaultSilentTokenRequest` (silent flow),  and `MSIDTokenResponseValidator` (interactive flow) won't change it. We still need to rely on `MSIDRequestParameters`/`MSIDConfiguration` to pass `v1IdTokenRequested=YES` to cache accessor such that it could construct the property cache key when save/find tokens.

But I am still ok adding those subclasses for both silent flow and interactive flow although i think it may introduce extra complexity :)

Reply:
What I meant is that we could use a subclass of MSIDDefaultTokenRequestProvider for ADAL based requests that would always request V1 id token type (we currently have a method in default accessor that looks up id token, we can extend it with id token type parameter). That way we don't need to have the v1IdTokenRequested parameter, because for ADAL requests we always need v1 id tokens and for MSAL requests we always need v2 id tokens.

Reply:
Addressed. The UT mocks have not been updated accordingly yet, as there might be more changes after review  (that's why UT is failing) :)